addons/html_editor/static/src/main/toolbar/toolbar_plugin.js

```diff
@@ -440,6 +440,10 @@ export class ToolbarPlugin extends Plugin {
      */
     updateButtonsStates(selection, targetedNodes) {
         const availableButtons = this.getAvailableButtonsSet(selection);
+        const disabledPredicates = this.getResource("toolbar_buttons_disabled_predicates");
```

Comment:
There is already an existingmechanism to disable toolbar buttons, it's the `isDisabled` property of `toolbar_buttons` resources.

Reply:
I'm available at GR2 tomorrow and Friday no problem if you want to discuss this.